### PR TITLE
draft two sync committee metrics

### PR DESCRIPTION
**Pull request by ybstaked**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

**Which issues(s) does this PR fix?**

Draft at fixing #11522 

**Other notes for review**

### CODE REVIEW

File: validator/client/sync_committee.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ func (v *validator) SubmitSyncCommitteeMessage(ctx context.Context, slot types.S
 		"validatorIndex":     msg.ValidatorIndex,
 	}).Info("Submitted new sync message")
 	atomic.AddUint64(&v.syncCommitteeStats.totalMessagesSubmitted, 1)
+
+	fmtKey := fmt.Sprintf("%#x", pubKey[:])
+	if v.emitAccountMetrics {
+		ValidatorSyncCommitteeGaugeVec.WithLabelValues(fmtKey).Set(float64(slot))
+	}
```
Comment (prestonvanloon) [inline validator/client/sync_committee.go:96]: I think there is a few problems with this approach.

1) This sets the value to a slot. I would assume, based on the description, that the gauge value would be a boolean 0 or 1.
2) This doesn't reset or unset the gauge when the validator stops being part of a current sync committee.
3) This doesn't fill all of your validator keys in the gauge. Same problem as number 2. 

For the third point, that may be desired, but I'm not sure. 

I think that we need to add more information to the function that reports validator performance metrics

File: validator/client/metrics.go [hunk 2]
```diff
@@ -190,6 +190,30 @@ var (
...
+		},
+		[]string{
+			"pubkey",
+		},
+	)
+	// ValidatorNextSyncCommitteeGaugeVec used to keep track of whether validator is part of next sync committee
+	ValidatorNextSyncCommitteeGaugeVec = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "validator",
+			Name:      "next_sync_committee",
```
Comment (prestonvanloon) [inline validator/client/metrics.go:209]: Naming wise, I would suggest something like 

- validator_sync_committee_current
- validator_sync_committee_next

The idea behind the naming is that it has a common prefix and makes it easier to find these metrics in some cases

**Comment by ybstaked** [comment]
This was just a draft I put out, indicating I want this exposed at the same metrics on the validator.  We run validator client as separate instance that connect to beacon node running as a separate instance.  Validator monitor service I think requires the operator to keep track and add keys to monitor dynamically rather than the validator metrics which are such exposed at the pubKeys it is managing level.

**Comment by potuz** [comment]
> This was just a draft I put out, indicating I want this exposed at the same metrics on the validator. We run validator client as separate instance that connect to beacon node running as a separate instance. Validator monitor service I think requires the operator to keep track and add keys to monitor dynamically rather than the validator metrics which are such exposed at the pubKeys it is managing level.

One of the items that we have in the todo is to dinamically add pubkeys to the validator monitor by adding a flag `--auto` that will automatically add those validators that are being served by the beacon. Would this work for you?

**Comment by ybstaked** [comment]
> 

so the validator monitor is exposing prometheus metrics at the beacon level?  I'd prefer the straightforward method of getting data on the VC metrics..

**Comment by potuz** [comment]
> > 
> 
> so the validator monitor is exposing prometheus metrics at the beacon level? I'd prefer the straightforward method of getting data on the VC metrics..

Yes, the validator monitor exposes metrics at the beacon level. Eventually we will deprecate and remove all metrics from the VC as they are covered by the beacon node with more.

**Comment by ybstaked** [comment]
> > > 
> > 
> > 
> > so the validator monitor is exposing prometheus metrics at the beacon level? I'd prefer the straightforward method of getting data on the VC metrics..
> 
> Yes, the validator monitor exposes metrics at the beacon level. Eventually we will deprecate and remove all metrics from the VC as they are covered by the beacon node with more.

OK, interesting, what is the consensus on getting standardization on these metrics approaches across the different client teams, prysm, teku, etc.?  Or is an approach like: https://github.com/samcm/ethereum-metrics-exporter more preferred??  This is important to align alongside the client diversity.  

But if the public plan is to deprecate the VC metrics, then I'm ok with this metric on the beacons for now.

**Pull request closed without merging**